src/shared/sddl.rs

```diff
@@ -0,0 +1,218 @@
+pub const SDDL_SPACEC: char = ' ';
+pub const SDDL_ACE_COND_BEGINC: char = '(';
+pub const SDDL_ACE_COND_ENDC: char = ')';
+pub const SDDL_ACE_COND_STRING_BEGINC: char = '"';
+pub const SDDL_ACE_COND_STRING_ENDC: char = '"';
+pub const SDDL_ACE_COND_COMPOSITEVALUE_BEGINC: char = '{';
+pub const SDDL_ACE_COND_COMPOSITEVALUE_ENDC: char = '}';
+pub const SDDL_ACE_COND_COMPOSITEVALUE_SEPERATORC: char = ',';
+pub const SDDL_ACE_COND_BLOB_PREFIXC: char = '#';
+pub const SDDL_ACE_COND_SID_BEGINC: char = '(';
+pub const SDDL_ACE_COND_SID_ENDC: char = ')';
+pub const SDDL_SEPERATOR: &'static str = ";";
+pub const SDDL_DELIMINATOR: &'static str = ":";
+pub const SDDL_ACE_BEGIN: &'static str = "(";
+pub const SDDL_ACE_END: &'static str = ")";
+pub const SDDL_ACE_COND_BEGIN: &'static str = "(";
+pub const SDDL_ACE_COND_END: &'static str = ")";
+pub const SDDL_SPACE: &'static str = " ";
+pub const SDDL_ACE_COND_BLOB_PREFIX: &'static str = "#";
+pub const SDDL_ACE_COND_SID_PREFIX: &'static str = "SID";
+pub const SDDL_ACE_COND_ATTRIBUTE_PREFIX: &'static str = "@";
+pub const SDDL_ACE_COND_USER_ATTRIBUTE_PREFIX: &'static str = "@USER.";
+pub const SDDL_ACE_COND_RESOURCE_ATTRIBUTE_PREFIX: &'static str = "@RESOURCE.";
+pub const SDDL_ACE_COND_DEVICE_ATTRIBUTE_PREFIX: &'static str = "@DEVICE.";
+pub const SDDL_ACE_COND_TOKEN_ATTRIBUTE_PREFIX: &'static str = "@TOKEN.";
+extern "system" {
+    pub fn ConvertSidToStringSidA(
+        Sid: PSID,
+        StringSid: *mut LPSTR,
+    ) -> BOOL;
+    pub fn ConvertSidToStringSidW(
+        Sid: PSID,
+        StringSid: *mut LPWSTR,
+    ) -> BOOL;
+    pub fn ConvertStringSidToSidA(
+        StringSid: LPCSTR,
+        Sid: *mut PSID,
+    ) -> BOOL;
+    pub fn ConvertStringSidToSidW(
+        StringSid: LPCWSTR,
+        Sid: *mut PSID,
+    ) -> BOOL;
+    pub fn ConvertStringSecurityDescriptorToSecurityDescriptorA(
+        StringSecurityDescriptor: LPCSTR,
+        StringSDRevision: DWORD,
+        SecurityDescriptor: *mut PSECURITY_DESCRIPTOR,
+        SecurityDescriptorSize: PULONG,
+    ) -> BOOL;
+    pub fn ConvertStringSecurityDescriptorToSecurityDescriptorW(
+        StringSecurityDescriptor: LPCWSTR,
+        StringSDRevision: DWORD,
+        SecurityDescriptor: *mut PSECURITY_DESCRIPTOR,
+        SecurityDescriptorSize: PULONG,
+    ) -> BOOL;
+    pub fn ConvertSecurityDescriptorToStringSecurityDescriptorA(
+        SecurityDescriptor: PSECURITY_DESCRIPTOR,
+        RequestedStringSDRevision: DWORD,
+        SecurityInformation: SECURITY_INFORMATION,
+        StringSecurityDescriptor: *mut LPSTR,
+        StringSecurityDescriptorLen: PULONG,
+    ) -> BOOL;
+    pub fn ConvertSecurityDescriptorToStringSecurityDescriptorW(
+        SecurityDescriptor: PSECURITY_DESCRIPTOR,
+        RequestedStringSDRevision: DWORD,
+        SecurityInformation: SECURITY_INFORMATION,
+        StringSecurityDescriptor: *mut LPWSTR,
+        StringSecurityDescriptorLen: PULONG,
+    ) -> BOOL;
+}
```

Comment:
Does Github ignore EOF newlines? It was missing before, but it's definitely there now. Checked the file directly downloaded from Github in a hex editor to make sure I wasn't going crazy.